src/main/java/com/sandogh/sandogh/users/service/UserService.java

```diff
@@ -16,8 +24,29 @@
 public class UserService {
     private final UserDAO userDAO;
 
+    public List<User> getAllUser(){
+        return (List<User>) userDAO.findAll();
+    }
+
+    public void getDeleteUser(long id){
+        this.userDAO.deleteById(id);
+    };
+
+    public User getUserById(long id){
+        Optional<User> optional=userDAO.findById(id);
+        User user=null;
+        if(optional.isPresent()){
+            user=optional.get();
+        }else
+        {
+            throw new RuntimeException("user not found for id ::" + id);
+        }
+        return user;
+    }
+
     private final Object CREATE_USER_LOCK = new Object();
 
+    //public User createNewUser(UserDTO user);
     public User createNewUser(String username, String password, String email, String phoneNumber) throws UsernameAlreadyTakeException {
```

Comment:
Check the TODO comments:
Apply all the validations regardless of what has been done on the other layers. DO NOT trust any unknown object that is passed to the Service methods.

Reply:
BTW, the final code should not have any TODO comments. Remove them after you applied your changes.